### PR TITLE
fix(migrate-config): robust handling of production bot.yaml shapes

### PR DESCRIPTION
**Pull request by jcfischer**
## Summary

MIG-7.9 dry-run against real grove-v2 bot.yaml on clawbox surfaced two schema-divergence crashes the existing migrator didn't handle. Both fixed defensively — cortex-shaped inputs pass through unchanged.

### 1. `trustedAgentBots[]` with Discord id but no symbolic name

**Production shape** (e.g. clawbox `~/.config/grove/bot.yaml`):
```yaml
trustedAgentBots:
  - id: \"1487180524542890144\"      # Discord snowflake, NO \`name:\` field
    role: \"agent-restricted\"
```

**Previously:** `Error: conversion failed: undefined is not an object (evaluating 'legacyName.trim')` — crash inside `buildTrustList` → `deriveAgentId(undefined)`.

**Fix:**
- Tolerate missing `name`: skip with warning, surface the raw platform id + role in the warning so the operator can hand-map the entry into the relevant agent's `trust:` list post-migration.
- Accept `id` as an alias for `discordId` (the production shape).

**Warning shape:**
```
trustedAgentBots[0] missing symbolic \`name\` (platform id: 1487180524542890144 role=\"agent-restricted\") —
  skipping. Hand-map this entry in cortex.yaml under the relevant agent's \`trust:\` list once you
  know which agent id corresponds to 1487180524542890144.
```

### 2. `nats.identity` in legacy `{did, keyPath}` shape

**Production shape:**
```yaml
nats:
  url: \"nats://127.0.0.1:4222\"
  identity:
    did: did:mf:jc-ivy
    keyPath: ~/.config/metafactory/keys/jc-ivy.key
```

**Cortex schema** requires `{seedPath, publicKey}` where `publicKey` is a 56-char U-prefix NKey user identifier.

**Previously:** crashed in `CortexConfigSchema.parse` with two zod `invalid_type` errors at `nats.identity.seedPath` and `nats.identity.publicKey`.

**Fix:** strip the legacy block with a warning that surfaces a concrete derivation hint:
```
legacy nats.identity shape (keys: did, keyPath; did=did:mf:jc-ivy) does not match cortex schema
  {seedPath, publicKey} — stripping block. Derive the U-prefixed pubkey via
  \`nkeys -inkey ~/.config/metafactory/keys/jc-ivy.key -pubout\` and add
  \`nats.identity.seedPath: ~/.config/metafactory/keys/jc-ivy.key\` +
  \`nats.identity.publicKey: U…\` to cortex.yaml.
```

The shapes can't be auto-translated — cortex needs the U-prefixed pubkey derived from the seed, which we don't have at migration time. Identity is optional in cortex; stripping with a hint is the right delegation.

## End-to-end validation

Dry-ran against clawbox's real bot.yaml (390 lines, 4 discord instances, 6 trustedAgentBots entries with only Discord ids, legacy `{did, keyPath}` nats identity). Output:

```
operator: jc (dataResidency=NZ)
agents:   4
  - ivy (discord) persona=./personas/ivy.md trust=[]
  - ivy-2 (discord) persona=./personas/ivy.md trust=[]
  - ivy-3 (discord) persona=./personas/ivy.md trust=[]
  - ivy-4 (discord) persona=./personas/ivy.md trust=[]
renderers: 1
  - dashboard

instance mappings:
  discord[0] discord-ivy     →  agents[ivy].presence.discord
  discord[1] discord-holly   →  agents[ivy-2].presence.discord
  discord[2] discord-juniper →  agents[ivy-3].presence.discord
  discord[3] discord-pilot   →  agents[ivy-4].presence.discord

warnings (10): 6 trustedAgentBots skipped + 1 persona-not-found +
               1 multi-instance + 1 api→dashboard + 1 nats.identity stripped
```

All 10 warnings are expected — the operator-side post-migration steps (hand-map 6 trust entries to agent ids, add `nats.identity` with derived pubkey, optionally rename `ivy-2..ivy-4` to `holly/juniper/pilot`) are surfaced clearly enough that the migration is now actionable.

## Tests

- 59/59 green in `migrate-config.test.ts` (was 55).
- 4 new tests for trustedAgentBots robustness (skip-with-warning, role-hint, mixed list, empty `name` string).
- 4 new tests for nats.identity (legacy strip + cortex pass-through + no-identity no-warn + garbage-shape hint omission).

## Test plan

- [x] `bunx tsc --noEmit` clean
- [x] `bun test src/cli/cortex/commands/__tests__/migrate-config.test.ts` — 59/59 green
- [x] End-to-end dry-run against clawbox bot.yaml: completes with valid cortex.yaml output + 10 expected warnings
- [ ] Live full-suite re-run post-merge (no expected regressions; touch surface is migrate-config-lib only)

## Out of scope

- **No auto-translation of legacy `{did, keyPath}` → `{seedPath, publicKey}`.** We don't have the public key at migration time (it's derived from the seed file we'd have to read + run through nkeys). The hint in the warning gives the operator the exact one-line command to derive it.
- **No agent-instance-name reattribution.** The migrator emits `ivy, ivy-2, ivy-3, ivy-4` per the existing variant-id convention. The operator will hand-rename to `ivy, holly, juniper, pilot` post-migration (each was a separate instance in grove-v2's flat `discord[]` array; cortex's schema reasonably treats them as numbered variants of the declared agent.name).

Unblocks MIG-7.9 (run migrate-config against real bot.yaml → emit cortex.yaml). Critical-path item for the Grove→Cortex switch.

### CODE REVIEW

File: src/cli/cortex/commands/__tests__/migrate-config.test.ts [hunk 1]
```diff
@@ -181,6 +181,148 @@ describe("convertBotYaml — trustedAgentBots", () => {
...
+  // skips those with a warning that surfaces the platform id so the operator
+  // can hand-map post-migration. Previously this path crashed with
+  // `undefined is not an object (evaluating 'legacyName.trim')`.
+  test("skips entries missing `name` with a warning naming the Discord id", () => {
+    const legacy: LegacyBotYaml = {
+      agent: { name: "ivy", displayName: "Ivy", operatorId: "jc" },
+      discord: [{ token: "t", guildId: "1", agentChannelId: "2", logChannelId: "3" }],
+      trustedAgentBots: [
+        // production shape — id + role only, no symbolic name
+        ({ id: "1487180524542890144", role: "agent-restricted" } as unknown) as LegacyBotYaml["trustedAgentBots"] extends (infer U)[] | undefined ? U : never,
```
Comment (jcfischer) [inline src/cli/cortex/commands/__tests__/migrate-config.test.ts:196]: **[nit/quality]** These `as unknown as LegacyBotYaml["trustedAgentBots"] extends (infer U)[] | undefined ? U : never` casts are vestigial. Now that `LegacyTrustedAgentBot` has `id?`, `role?`, and `name?` all optional, `{ id: "1487180524542890144", role: "agent-restricted" }` already satisfies the type directly. The conditional-infer gymnastics add 100 chars of noise per line for zero type safety. Replace with plain objects — or at most `as LegacyTrustedAgentBot`.

**Comment by jcfischer** [comment]
## Full Review: the-metafactory/cortex#82

### Lenses Applied (6 + duplication)
1. CodeQuality — pass (1 finding)
2. Security — pass (0 findings)
3. Hardening — pass (0 findings — not applicable: pure function, no API surface)
4. Architecture — pass (0 findings)
5. EcosystemCompliance — pass (0 findings)
6. Performance — pass (0 findings)
7. Duplication — pass (0 findings)

### Lens Results
| Lens | Critical | Warning | Suggestion | Nit | Verdict |
|------|----------|---------|------------|-----|---------|
| CodeQuality | 0 | 0 | 0 | 1 | pass |
| Security | 0 | 0 | 0 | 0 | pass |
| Hardening | 0 | 0 | 0 | 0 | pass |
| Architecture | 0 | 0 | 0 | 0 | pass |
| EcosystemCompliance | 0 | 0 | 0 | 0 | pass |
| Performance | 0 | 0 | 0 | 0 | pass |
| Duplication | 0 | 0 | 0 | 0 | pass |

### Key Observations

1. **The fix is correct and minimal.** Both crash paths (missing `name` in trustedAgentBots, legacy `{did, keyPath}` nats.identity) are now handled defensively with actionable operator warnings. The skip-and-warn approach is the right call — auto-translation isn't possible for either case.

2. **Test coverage is thorough.** 8 new tests cover: missing name, discordId fallback, mixed lists, whitespace-only name, legacy identity strip, cortex-shape passthrough, missing identity, and garbage-shape identity. Edge cases I checked for and found covered.

3. **`convertNats` returns `unknown` but the output is validated.** The untyped return gets passed through `CortexConfigSchema.parse()` at the end of `convertBotYaml`, so runtime safety is enforced by the Zod schema. Acceptable tradeoff — typing it more precisely would require re-declaring the NatsConfig shape minus identity, which buys nothing since Zod validates it anyway.

4. **Interface change (`name` now optional) is non-breaking.** `LegacyTrustedAgentBot` is an input type consumed only by `convertBotYaml`. Making `name` optional widens the input contract — all existing callers still compile. Adding `id?` and `role?` is additive.

5. **59/59 tests green locally.** Confirmed by running the suite against the PR branch.

### Overall Verdict: approve
Clean defensive fix. One nit on vestigial type casts in tests (see inline). No blockers, no majors.

File: src/cli/cortex/commands/__tests__/migrate-config.test.ts (hunk 1 above)
Comment (jcfischer) [inline src/cli/cortex/commands/__tests__/migrate-config.test.ts:196]: **[nit/quality]** With `name` now optional on `LegacyTrustedAgentBot`, objects like `{ id: "1487180524542890144", role: "agent-restricted" }` already satisfy the interface. The `(... as unknown) as LegacyBotYaml["trustedAgentBots"] extends (infer U)[] | undefined ? U : never` coercion is unnecessary boilerplate — just assign the literal directly.

Three occurrences in this block.

**Review by jcfischer — COMMENTED** [review]
Full review (6 lenses + duplication) — clean defensive fix for two production bot.yaml shape crashes. Both handled with skip-and-warn, 8 new tests, 59/59 green.

verdict: blockers=0 majors=0 nits=1 — recommend: merge (posted as comment-review — bot account opened the PR; --approve blocked by GitHub)

**Comment by jcfischer** [comment]
## Full Review: the-metafactory/cortex#82

### Lens Results
| Lens | Critical | Warning | Suggestion | Nit | Verdict |
|------|----------|---------|------------|-----|---------|
| CodeQuality | 0 | 0 | 0 | 1 | pass |
| Security | 0 | 0 | 0 | 0 | pass |
| Hardening | — | — | — | — | N/A (pure lib) |
| Architecture | 0 | 0 | 0 | 0 | pass |
| EcosystemCompliance | 0 | 0 | 0 | 0 | pass |
| Performance | — | — | — | — | N/A (one-shot) |
| Duplication | 0 | 0 | 0 | 0 | pass |

### What I verified

- `buildTrustList`: all paths — named, empty-name, id-only, mixed lists. `.map()` → `for` loop is necessary for `continue` semantics. Ordering preserved. Throw on non-derivable names preserved.
- `convertNats`: null input (passthrough), cortex-shaped identity (passthrough), legacy `{did, keyPath}` (strip + derive-hint), garbage shape (strip + fallback hint), no identity block (passthrough). Shallow copy prevents input mutation.
- `hasCortexShape` gate correctly delegates full `NatsIdentitySchema` regex validation to the terminal `CortexConfigSchema.parse()`.
- `LegacyTrustedAgentBot.name: string → name?: string` is backward-compatible: existing callers still compile, `buildTrustList` handles `undefined` before reaching `deriveAgentId`.
- Existing test "passes nats config through unchanged" still works — `full.bot.yaml` has no identity block.
- 8 new tests cover the right edge cases. Warning messages carry actionable operator hints.
- No hidden coupling — `convertNats` follows the same per-block converter extraction pattern as `convertDiscordPresence`/`convertMattermostPresence`.

### Overall Verdict: approve
Clean defensive fix. One cosmetic nit (unnecessary type coercions in test fixtures) — not worth blocking a critical-path MIG-7.9 unblocker.

**Review by jcfischer — COMMENTED** [review]
Full review (6 lenses + duplication) — clean defensive fix for two production bot.yaml shape divergences. One nit (unnecessary type coercions in tests).

verdict: blockers=0 majors=0 nits=1 — recommend: merge (posted as comment-review — bot account opened the PR; --approve blocked by GitHub)
